### PR TITLE
Reconcile aws infra count. (#1490)

### PR DESCRIPTION
**Pull request by Ficus-f**
Feature: reconcile aws infra count. (#1490)

### CODE REVIEW

**Comment by fanux** [comment]
you need `git rebase upstream/main`

File: controllers/infra/drivers/aws/get_instance.go [hunk 1]
```diff
@@ -126,6 +126,9 @@ func (d Driver) getInstancesByLabel(key string, value string, infra *v1.Infra) (
 
 	for _, r := range result.Reservations {
 		for _, i := range r.Instances {
+			if i.State.Name == types.InstanceStateNameShuttingDown || i.State.Name == types.InstanceStateNameTerminated || i.State.Name == types.InstanceStateNameStopping || i.State.Name == types.InstanceStateNameStopped {
```
Comment (fanux) [inline controllers/infra/drivers/aws/get_instance.go:129]: Instances in the stopped or stopping state cannot be ignored. Shutting down and terminated will auto clean by aws

Comment (Ficus-f) [reply controllers/infra/drivers/aws/get_instance.go:129]: Terminated aws will cause an error if we want to get its PrivateIpAddress when the aws terminates but is not cleaned.

File: controllers/infra/drivers/aws/get_instance.go [hunk 2]
```diff
@@ -126,6 +126,9 @@ func (d Driver) getInstancesByLabel(key string, value string, infra *v1.Infra) (
...
+				continue
+			}
```
Comment (fanux) [inline controllers/infra/drivers/aws/get_instance.go:131]: if stop {
   start 
}

**Review by fanux — APPROVED** [review]

**Pull request merged**